patricia_trie/src/nibbleslice.rs

```diff
@@ -147,12 +152,29 @@ impl<'a> NibbleSlice<'a> {
  	/// How many of the same nibbles at the beginning do we match with `them`?
 	pub fn common_prefix(&self, them: &Self) -> usize {
 		let s = min(self.len(), them.len());
-		let mut i = 0usize;
-		while i < s {
-			if self.at(i) != them.at(i) { break; }
-			i += 1;
+		if s % 2 == 0 && s > 2 && self.offset % 2 == 0 && them.offset % 2 == 0 {
+			let my_offset = self.offset / 2;
+			let their_offset = them.offset / 2;
+			let mut eq_counter = 0;
+			for i in 0..s/2 {
+				let mybyte = self.data[i + my_offset];
+				let theirbyte = them.data[i + their_offset];
+				if mybyte == theirbyte { eq_counter += 2 } else {
+					if mybyte & 0b_1111_0000 == theirbyte & 0b_1111_0000 {
+						eq_counter += 1;
+					}
+					break
+				}
+			}
+			return eq_counter
```

Comment:
I'm not sure how it works, but this behavior seems different than it used to be if `!self.data_encode_suffix.empty()`. Or this can't be true if `if s % 2 == 0 && s > 2 && self.offset % 2 == 0 && them.offset % 2 == 0`?

Reply:
You are correct, totally missed the "composed" case, good thing you spotted this!